### ISSUE TITLE
[Bug]: message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=103, reason=ShardNotAvailable

### ISSUE THREAD

**Issue by Cheung7788** [milvus-io/milvus#22854]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

inser 400w entities to sample_data partition, query error：fail to Search, QueryNode ID=103, reason=ShardNotAvailable

### Milvus Log

[2023-03-20 10:54:51,318][decorators.py:108][handler][ThreadPoolExecutor-0_0:140059813607168][ERROR][pid:888322][RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=103, reason=ShardNotAvailable(replica=439746809171804528, shard=li-milvus-test-rootcoord-dml_242_439746689672782277v2))>, <Time:{'RPC start': '2023-03-20 10:54:50.406747', 'RPC error': '2023-03-20 10:54:51.318784'}>]
[2023-03-20 10:54:51,318][ann.py:100][search][ThreadPoolExecutor-0_0:140059813607168][ERROR][pid:888322][<MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=103, reason=ShardNotAvailable(replica=439746809171804528, shard=li-milvus-test-rootcoord-dml_242_439746689672782277v2))>]
--- Logging error ---
Traceback (most recent call last):
  File "/home/ops/workspace/image_search_pipeline/core/ann.py", line 98, in search
    search_res = self.collection_handler.search(vector_to_search, "embeddings", search_params, limit=topk, output_fields=output_fields, timeout=self.read_timeout, partition_names=partition)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/orm/collection.py", line 616, in search
    schema=self._schema_dict, **kwargs)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/client/grpc_handler.py", line 472, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/client/grpc_handler.py", line 441, in _execute_search_requests
    raise pre_err
  File "/home/ops/anaconda3/envs/img2vec/lib/python3.7/site-packages/pymilvus/client/grpc_handler.py", line 432, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=103, reason=ShardNotAvailable(replica=439746809171804528, shard=li-milvus-test-rootcoord-dml_242_439746689672782277v2))>

### Anything else?

etcd_backup:
[bw_etcd_ALL.230320-120008.bak.gz](https://github.com/milvus-io/milvus/files/11015004/bw_etcd_ALL.230320-120008.bak.gz)
logs:
https://drive.google.com/drive/u/0/folders/1sAF3JkgJkB1vg_rqIEkb3h3bNXGEwGjt

**Comment by Cheung7788** [comment]
@wayblink

**Comment by HongqiZhao** [comment]
记个中间态，目前16:36集群已恢复正常。从10:00~16:36 中间没有什么特殊操作，就是和周末一样的插入，脚本服务没动，然后也有一些search。
具体什么时候恢复正常的不确定，中间异常已知持续时间大约有>2h。期望排查根因，后面发生好快速解决。

**Comment by wayblink** [comment]
One querynode failed and quit. Caused by "Exclusive consumer is already connected", similar to #20864. 

This issue has been partly solved(lower possibility) in new version(2.2.3+).

Rebalance data to new querynode, it costs a long time, 2.5h. 

High recommend to update to newer version 2.2.3+
